beacon-chain/rpc/validator_server.go

```diff
@@ -410,51 +422,6 @@ func (vs *ValidatorServer) lookupValidatorStatus(validatorIdx uint64, beaconStat
 	return status
 }
 
-// filterActivePublicKeys takes a list of validator public keys and returns
-// the list of active public keys from the given state.
-func (vs *ValidatorServer) filterActivePublicKeys(beaconState *pbp2p.BeaconState, pubkeys [][]byte) [][]byte {
-	// Generate a map for O(1) lookup of existence of pub keys in request.
-	pkMap := make(map[string]bool)
-	for _, pk := range pubkeys {
-		pkMap[hex.EncodeToString(pk)] = true
-	}
-
-	var activeKeys [][]byte
-	currentEpoch := helpers.SlotToEpoch(beaconState.Slot)
-	for _, v := range beaconState.ValidatorRegistry {
```

Comment:
this is very expensive, for every assignment request we have to loop through the entire validator list just to get the active public keys